File: server/datasets/java_source-code/binary_search_tree/type_2_renamed2.java
class Node {
    int val;
    Node left, right;

    Node(int val) {
        this.val = val;
        left = right = null;
    }
}

class BST {
    Node root;

    public void insertValue(int value) {
        root = insert(root, value);
    }

    private Node insert(Node current, int value) {
        if (current == null) {
            return new Node(value);
        }

        if (value < current.val) {
            current.left = insert(current.left, value);
        } else {
            current.right = insert(current.right, value);
        }

        return current;
    }

    public boolean contains(int value) {
        return search(root, value);
    }

    private boolean search(Node node, int value) {
        if (node == null) return false;

        if (node.val == value) return true;

        if (value < node.val) return search(node.left, value);

        return search(node.right, value);
    }
}

File: server/datasets/python_source-code/binary_search_tree/type_2_renamed2.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None


class BST:
    def __init__(self):
        self.root = None

    def insert_value(self, value):
        self.root = self._insert(self.root, value)

    def _insert(self, current, value):
        if current is None:
            return Node(value)

        if value < current.val:
            current.left = self._insert(current.left, value)
        else:
            current.right = self._insert(current.right, value)

        return current

    def contains(self, value):
        return self._search(self.root, value)

    def _search(self, node, value):
        if node is None:
            return False

        if node.val == value:
            return True

        if value < node.val:
            return self._search(node.left, value)

        return self._search(node.right, value)
